js/task17.js

```diff
@@ -0,0 +1,129 @@
+const ul = document.getElementById('lists');
+
+let currentImage = 0;
+const imagesArr = [];
+
+function createLoader() {
+	const fragment = document.createDocumentFragment();
+	const loaderContainer = document.createElement('li');
+	const loader = document.createElement('img');
+	loader.src = './img/loading-circle.gif';
+	fragment.appendChild(loader);
+	loader.appendChild(loaderContainer);
+	loader.classList.add('loader_image');
+	return loader;
+}
+
+function fetchData() {
+	return new Promise((resolve, reject) => {
+		setTimeout(() => {
+			resolve(fetch('./js/json/task17.json'));
+		}, 3000);
+	});
+}
+
+async function getData() {
+	const loaderImage = createLoader();
+
+	const fragment = document.createDocumentFragment();
+	const li = document.createElement('li');
+	li.classList.add('loader');
+
+	li.appendChild(loaderImage);
+	fragment.appendChild(li);
+	console.log('fragment', fragment);
+	ul.appendChild(fragment);
+
+	try {
+		const response = await fetchData();
+		const res = await response.json();
+		const data = res.data;
+
+		data.map((el) => {
+			const image = document.createElement('img');
+			const li = document.createElement('li');
+
+			image.src = el.image;
+			li.appendChild(image);
+
+			imagesArr.push(li);
+		});
+
+		imagesArr.forEach((el, index) => {
+			el.classList.add('slide-images');
+
+			if (index === 0) {
+				el.classList.add('z-index-100');
+			}
+			ul.appendChild(el);
+		});
+	} catch (error) {
+		ul.innerHTML = "Couldn't get response.";
+	} finally {
+		console.log('You got the result.');
+	}
+}
+getData();
+
+function displayImage(currentImage, clickEvent) {

```

Comment:
Okay, I'll separate those functions. I just combined them because I was using almost the same code in both functions. I'll try to figure out a better solution.